test/Microsoft.IdentityModel.Tokens.Tests/CrossTokenTests.cs

```diff
@@ -102,6 +103,123 @@ public static TheoryData<CrossTokenTheoryData> CrossTokenValidateTokenTheoryData
             }
         }
 
+        [Theory, MemberData(nameof(CanReadTokenTheoryData))]
+        public void CanReadToken(TokenHandlerTheoryData theoryData)
```

Comment:
I really love this idea of CrossToken tests. This will help ensure that from the SecurityTokenHandler view, they work the same.